api-compact.yaml

```diff
@@ -66,6 +59,41 @@ apigateway_responses_with_mapping_template: &apigateway_responses_with_mapping_t
   500:
     statusCode: "500"
 
+apigateway_catalog_courses_response_with_mapping_template: &apigateway_catalog_courses_response_with_mapping_template
+  default:
+    statusCode: "400"
+  200:
+    statusCode: "200"
+    responseTemplates:
+      # Response body mapping template, this template is used for updating
+      # "next" and "previous" urls (both host and url path) while preserving
+      # the protocol (http | https) and querystring parameters.
+      application/json: >
+        #set($inputRoot = $input.path('$'))
+        #set($host = $stageVariables.gateway_host)
+        #set($catalogId = $input.params('id'))
```

Comment:
@douglashall `$context.resourceId ` does not contain catalog id.

ref: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html#context-variable-reference
> $context.resourceId : The identifier API Gateway assigns to your resource.